spec/json_matchers/match_json_schema_spec.rb

```diff
@@ -59,6 +59,147 @@
     expect(json).to match_json_schema("api/v1/schema")
   end
 
+  it "supports invalidating the referenced schema when using local references" do
+    create(:schema, name: "post", json: {
+      "$schema": "https://json-schema.org/draft-04/schema#",
+      "id": "file:/post.json#",
+      "definitions": {
+        "attributes": {
+          "type": "object",
+          "required": [
+            "id",
+            "name",
+          ],
+          "properties": {
+            "id": { "type": "string" },
+            "name": { "type": "string" }
+          }
+        }
+      },
+      "type": "object",
+      "required": [
+        "id",
+        "type",
+        "attributes"
+      ],
+      "properties": {
+        "id": { "type": "string" },
+        "type": { "type": "string" },
+        "attributes": {
+          "$ref": "#/definitions/attributes",
+        }
+      }
+    })
+    posts_index = create(:schema, name: "posts/index", json: {
+      "$schema": "https://json-schema.org/draft-04/schema#",
+      "id": "file:/posts/index.json#",
+      "type": "object",
+      "required": [
+        "data"
+      ],
+      "definitions": {
+        "posts": {
+          "type": "array",
+          "items": {
+            "$ref": "file:/post.json#"
+          }
+        }
+      },
+      "properties": {
+        "data": {
+          "$ref": "#/definitions/posts"
+        }
+      }
+    })
+
+    json = build(:response, {
+      "data": [{
+        "id": 1,
```

Comment:
Convert this id attribute as a string, i.e. "id": "1". Keep the id inside the attributes as a number. The test will fail now.

Reply:
Aha! I've pushed upd2027df. Thanks for clarifying @musaffa !